homeassistant/components/netatmo/climate.py

```diff
@@ -3,71 +3,132 @@
+    STATE_NETATMO_OFF: STATE_OFF,
+    STATE_NETATMO_MANUAL: STATE_MANUAL
+}
+
+DICT_HA_TO_NETATMO = {
+    STATE_AUTO: STATE_NETATMO_SCHEDULE,
+    STATE_COOL: STATE_NETATMO_HG,
+    STATE_HEAT: STATE_NETATMO_MAX,
+    STATE_ECO: STATE_NETATMO_AWAY,
+    STATE_OFF: STATE_NETATMO_OFF,
+    STATE_MANUAL: STATE_NETATMO_MANUAL
+}
+
 SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE |
                  SUPPORT_AWAY_MODE)
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the NetAtmo Thermostat."""
     netatmo = hass.components.netatmo
-    device = config.get(CONF_RELAY)
 
     import pyatmo
+    homes_conf = config.get(CONF_HOMES)
     try:
-        data = ThermostatData(netatmo.NETATMO_AUTH, device)
-        for module_name in data.get_module_names():
-            if CONF_THERMOSTAT in config:
-                if config[CONF_THERMOSTAT] != [] and \
-                   module_name not in config[CONF_THERMOSTAT]:
-                    continue
-            add_entities([NetatmoThermostat(data, module_name)], True)
+        home_data = HomeData(netatmo.NETATMO_AUTH)
     except pyatmo.NoDevice:
-        return None
+        return
+
+    homes = []
+    rooms = {}
+    if homes_conf is not None:
+        for home_conf in homes_conf:
+            home = home_conf.get(CONF_NAME)
+            if CONF_ROOMS in home_conf and home_conf[CONF_ROOMS] != []:
+                rooms[home] = home_conf.get(CONF_ROOMS)
+            homes.append(home)
+    else:
+        homes = home_data.get_home_names()
+
+    for home in homes:
+        _LOGGER.debug("Setting up %s ...", home)
+        try:
+            room_data = ThermostatData(netatmo.NETATMO_AUTH, home)
+        except pyatmo.NoDevice:
+            continue
+        for room_id in room_data.get_room_ids():
+            room_name = room_data.homedata.rooms[home][room_id]['name']
+            _LOGGER.debug("Setting up %s (%s) ...", room_name, room_id)
+            if home in rooms and room_name not in rooms[home]:
+                _LOGGER.debug("Excluding %s ...", room_name)
+                continue
+            _LOGGER.debug("Adding devices for room %s (%s) ...",
+                          room_name, room_id)
+            add_entities([NetatmoThermostat(room_data, room_id)], True)
 
 
 class NetatmoThermostat(ClimateDevice):
     """Representation a Netatmo thermostat."""
 
-    def __init__(self, data, module_name, away_temp=None):
+    def __init__(self, data, room_id):
         """Initialize the sensor."""
         self._data = data
         self._state = None
-        self._name = module_name
+        self._room_id = room_id
+        room_name = self._data.homedata.rooms[self._data.home][room_id]['name']
+        self._name = 'netatmo_' + room_name
         self._target_temperature = None
         self._away = None
+        self._module_type = self._data.room_status[room_id]['module_type']
+        self._support_flags = SUPPORT_FLAGS
```

Comment:
Rather than overriding these with the if statement below, I think it would be cleaner to have an if/else here.

Reply:
Resolved in 4515f57.